samza-test/src/test/java/org/apache/samza/test/operator/TestRepartitionJoinWindowApp.java

```diff
@@ -76,6 +84,27 @@ public void testRepartitionJoinWindowApp() throws Exception {
       Assert.assertTrue(key.equals("u1") || key.equals("u2"));
       Assert.assertEquals("2", value);
     }
+
+    // Verify that messages in the intermediate stream will be deleted in 10 seconds
```

Comment:
I would also add a negative test to ensure that if `deleteMessages` is set to false, we are able to consume messages from the intermediate stream.

Reply:
Sure. I will add the negative test.

Reply:
I added a negative test but it is not easy to run it together with the positive test. These two tests interactive with each other because they use the same Kafka topics.

I can refactor `TestRepartitionJoinWindowApp` and `RepartitionJoinWindowApp` so that it uses different topics for different runs. Not sure if it is needed. What do you think?